lib/sample_uploader/utils/sesar_api.py

```diff
@@ -0,0 +1,61 @@
+import requests
+
+"""
+SESAR REST web service API
+
+This API handles web request to a REST web service provided by SESAR (the System for Earth Sample Registration)
+
+The official SESAR documentation about this web service can be found here:
+https://www.geosamples.org/interop
+
+"""
+
+__author__ = "Tianhao Gu <tgu@anl.gov>"
+
+
+def _get_igsn_endpoint():
+    """
+    Retrieve endpoint for getting sample profile for specific IGSN
+
+    The official SESAR documentation about this endpoint:
+    https://www.geosamples.org/interop#Sample-profile-IGSN
+    """
+    return 'https://app.geosamples.org/sample/igsn'
+
+
+def retrieve_sample(igsn):
+    """
+    Retrieve sample profile for given IGSN
+
+    The official SESAR documentation about this endpoint:
+    https://www.geosamples.org/interop#Sample-profile-IGSN
+    """
+
+    return_format = 'application/json'
+    headers = {"accept": return_format}
+
+    url = '{}/{}'.format(_get_igsn_endpoint(), igsn)
+
+    resp = requests.get(url=url, headers=headers)
```

Comment:
May want to catch exceptions here. See https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module

Reply:
👍